api_v3/lib/types/resource/KalturaEntryResource.php

```diff
@@ -51,15 +51,11 @@ public function validateForUsage($sourceObject, $propertiesToSkip = array())
 		
 		$key = $srcFlavorAsset->getSyncKey(asset::FILE_SYNC_ASSET_SUB_TYPE_ASSET);
 		$c = FileSyncPeer::getCriteriaForFileSyncKey($key);
-		$c->addAnd(FileSyncPeer::FILE_TYPE, array(FileSync::FILE_SYNC_FILE_TYPE_FILE, FileSync::FILE_SYNC_FILE_TYPE_LINK), Criteria::IN);
 		 
 		$fileSyncs = FileSyncPeer::doSelect($c);
-		foreach($fileSyncs as $fileSync)
-		{
-			$fileSync = kFileSyncUtils::resolve($fileSync);
-			if($fileSync->getFileType() == FileSync::FILE_SYNC_FILE_TYPE_FILE)
```

Comment:
This test is still needed, check that the file-type is not link after the resolve.